geth/peers/peerpool.go

```diff
@@ -182,8 +182,10 @@ func (p *PeerPool) handleServerPeers(server *p2p.Server, events <-chan *p2p.Peer
 			p.stopDiscovery(server)
 			return
 		case <-timeout:
-			log.Debug("DiscV5 timed out", "server", server.Self())
-			p.stopDiscovery(server)
+			if p.stopOnMax {
```

Comment:
it probably needs to be handled one step higher, so that timeout will be actually always nil chan if p.stopOnMax is true